### PR TITLE
OAK-10724 : Introduce detailed gc mode that only deletes orphan nodes…

### PR DESCRIPTION
**Pull request by stefan-egli**
… and deleted properties

### CODE REVIEW

**Comment by stefan-egli** [comment]
BranchCommitGCTest now fail due to using the new mode by default - requires fixing first before merging this

**Review by Joscorbe — APPROVED** [review]

File: oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java
Comment (rishabhdaim) [inline oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java:1012]: I would add an INFO log here.

**Review by rishabhdaim — APPROVED** [review]

**Comment by rishabhdaim** [comment]
PR approved with a minor review comment that can be handled later.

File: oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java
Comment (stefan-egli) [reply oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java:1012]: this would log it quite frequently no? - what about doing it in the constructor?

Comment (stefan-egli) [reply oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java:1012]: (I'll merge without and we can do it separately as you suggested)

**Comment by stefan-egli** [comment]
> BranchCommitGCTest now fail due to using the new mode by default - requires fixing first before merging this

will be handled in OAK-10726 now (to not block this PR)

log.info we need to follow-up in a new ticket still

File: oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java
Comment (stefan-egli) [reply oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java:1012]: PS: created OAK-10727 to follow-up on this

**Pull request merged**
